### PR TITLE
[SPARK-41334][CONNECT][PYTHON] Move `SortOrder` proto from relations to expressions

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?

1. Move `SortOrder` proto from relations to expressions
2. rename fields according to the Expression in Scala

### Why are the changes needed?
`SortOrder` is a common expression, not just in `Sort`, it will also be used in `Window` and `repartitionByRange`

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
updated UT

### CODE REVIEW

**Review by HyukjinKwon — APPROVED** [review]

**Review by amaliujia — APPROVED** [review]
LGTM

File: python/pyspark/sql/connect/column.py [hunk 1]
```diff
@@ -437,35 +437,38 @@ def to_plan(self, session: "SparkConnectClient") -> proto.Expression:
...
 
-        if self._nullsLast:
-            sort.nulls = proto.Sort.SortNulls.SORT_NULLS_LAST
+        if self._nullsFirst:
+            sort.sort_order.null_ordering = proto.Expression.SortOrder.NullOrdering.SORT_NULLS_FIRST
         else:
-            sort.nulls = proto.Sort.SortNulls.SORT_NULLS_FIRST
+            sort.sort_order.null_ordering = proto.Expression.SortOrder.NullOrdering.SORT_NULLS_LAST
 
-        return cast(proto.Expression, sort)
```
Comment (dongjoon-hyun) [inline python/pyspark/sql/connect/column.py:468]: This seems to cause `unused import` linter failure. Could you fix it?
```
./python/pyspark/sql/connect/column.py:18:1: F401 'typing.cast' imported but unused
from typing import (
     F401 'typing.cast' imported but unused
```

File: connector/connect/common/src/main/protobuf/spark/connect/relations.proto [hunk 2]
```diff
@@ -246,29 +246,11 @@ message Sort {
   // (Required) Input relation for a Sort.
   Relation input = 1;
 
-  // (Required) Sort fields.
-  repeated SortField sort_fields = 2;
+  // (Required) The ordering expressions
```
Comment (grundprinzip) [inline connector/connect/common/src/main/protobuf/spark/connect/relations.proto:249]:
```suggestion
  // (Required) The ordering expressions.
```

File: connector/connect/common/src/main/protobuf/spark/connect/expressions.proto [hunk 3]
```diff
@@ -37,6 +37,26 @@ message Expression {
     Alias alias = 6;
     Cast cast = 7;
     UnresolvedRegex unresolved_regex = 8;
+    SortOrder sort_order = 9;
+  }
+
+  message SortOrder {
```
Comment (grundprinzip) [inline connector/connect/common/src/main/protobuf/spark/connect/expressions.proto:43]: doc please

File: connector/connect/common/src/main/protobuf/spark/connect/expressions.proto [hunk 4]
```diff
@@ -37,6 +37,26 @@ message Expression {
     Alias alias = 6;
     Cast cast = 7;
     UnresolvedRegex unresolved_regex = 8;
+    SortOrder sort_order = 9;
```
Comment (grundprinzip) [inline connector/connect/common/src/main/protobuf/spark/connect/expressions.proto:40]: Do you mind adding a bit of documentation that would explain what are valid uses of `SortOrder` in expressions? This is just to make sure it's understood that you can't use this in a projection etc.

If the only use-case for SortOrder is to be used in `Sort` relation and `Window` then maybe it's better to pull it out from the expressions message.

File: python/pyspark/sql/connect/column.py (hunk 1 above)
Comment (zhengruifeng) [reply python/pyspark/sql/connect/column.py:468]: thanks, will update

File: connector/connect/common/src/main/protobuf/spark/connect/expressions.proto (hunk 4 above)
Comment (zhengruifeng) [reply connector/connect/common/src/main/protobuf/spark/connect/expressions.proto:40]: I add a note for not support in projection, but I guess the developers should be aware of this.

File: connector/connect/common/src/main/protobuf/spark/connect/expressions.proto (hunk 3 above)
Comment (zhengruifeng) [reply connector/connect/common/src/main/protobuf/spark/connect/expressions.proto:45]: done

**Comment by zhengruifeng** [comment]
merged into master, thanks for reivews

**Pull request closed without merging**